components/ruby/lib/chef-licensing/license_key_fetcher.rb

```diff
@@ -71,7 +71,7 @@ def perform_on_prem_operations
 
       unless @license_keys.empty?
         # Licenses expiration check
-        if licenses_active?
+        if licenses_active? || client_api_call_error
```

Comment:
why would we want to return license keys if we have client api call error as true?

Reply:
I will remove this check from this PR. Have created a separate PR explaining this flag usage https://github.com/chef/chef-licensing/pull/162